acceptance/launcher_test.go

```diff
@@ -0,0 +1,215 @@
+package acceptance
+
+import (
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/sclevine/spec"
+	"github.com/sclevine/spec/report"
+
+	h "github.com/buildpacks/lifecycle/testhelpers"
+)
+
+var (
+	launchDockerContext = filepath.Join("testdata", "launcher")
+	launcherBinaryDir   = filepath.Join("acceptance", "testdata", "launcher", "container", "cnb", "lifecycle")
+	launchImage         = "lifecycle/acceptance/launcher"
+)
+
+func TestLauncher(t *testing.T) {
+	buildLauncher(t)
+	buildLaunchImage(t)
+	defer removeLaunchImage(t)
+	spec.Run(t, "acceptance", testLauncher, spec.Parallel(), spec.Report(report.Terminal{}))
+}
+
+func testLauncher(t *testing.T, when spec.G, it spec.S) {
+	when("no args and no default", func() {
+		it("launches web", func() {
+			cmd := exec.Command("docker", "run", "--rm", launchImage)
+			output, err := cmd.CombinedOutput()
+			if err != nil {
+				t.Fatalf("failed to run %v\n OUTPUT: %s\n ERROR: %s\n", cmd.Args, output, err)
+			}
+			expected := "Executing web process-type"
+			if !strings.Contains(string(output), expected) {
+				t.Fatalf("failed to execute web:\n\t got: %s\n\t want: %s", output, expected)
+			}
+		})
+	})
+
+	when("no args and CNB_PROCESS_TYPE", func() {
+		it("launches given default process", func() {
+			cmd := exec.Command("docker", "run", "--rm", "--env", "CNB_PROCESS_TYPE=other-process", launchImage)
+			output, err := cmd.CombinedOutput()
+			if err != nil {
+				t.Fatalf("failed to run %v\n OUTPUT: %s\n ERROR: %s\n", cmd.Args, output, err)
+			}
+			expected := "Executing other-process process-type"
+			if !strings.Contains(string(output), expected) {
+				t.Fatalf("failed to execute other-process:\n\t got: %s\n\t want: %s", output, expected)
+			}
+		})
+	})
+
+	when("process provided in CMD", func() {
+		it("launches process", func() {
+			cmd := exec.Command("docker", "run", "--rm", launchImage, "other-process")
+			output, err := cmd.CombinedOutput()
+			if err != nil {
+				t.Fatalf("failed to run %v\n OUTPUT: %s\n ERROR: %s\n", cmd.Args, output, err)
+			}
+			expected := "Executing other-process process-type"
+			if !strings.Contains(string(output), expected) {
+				t.Fatalf("failed to execute other-process:\n\t got: %s\n\t want: %s", output, expected)
+			}
+		})
+	})
+
+	when("CMD provided", func() {
+		it("sources profiles and launches command with shell", func() {
+			cmd := exec.Command("docker", "run", "--rm", launchImage, "echo my-cmd")
```

Comment:
I wonder if we can change `my-cmd` to something else - this name makes me think that `my-cmd` is a fake command that we have a fixture for...